### DISCUSSION TITLE
"times" edge case

### DISCUSSION THREAD

**Discussion by DanielAjoy** [h3rald/min#175] [General]
While trying to define a function that creates a sequence of numbers provided a length of the list, I noticed that **times** does not accept 0 as a number of times to repeat a quote. I'd argue that repeating something 0 times is perfectly valid and useful:

(empty swap dup (pred dup (swons) dip) swap times pop) ^iota

```
$ 2 iota
 (
   0
   1
  )
```
```
$ 10 iota
 (
   0
   1
   2
   3
   4
   5
   6
   7
   8
   9
  )
```

Here is the problem:

```
$ 0 iota
(!) <repl>(1,49) [times]: A non-zero natural number is required
    <repl>(1,49) in symbol: times
    <repl>(1,6) in symbol: iota
```

Was expecting the empty list as the result

```
0 iota
()
```

**Comment by DanielAjoy** [comment]
Sorry **empty** was defined this way

`((()) apply) ^empty`

**Comment by h3rald** [comment]
Yeah makes sense. Plus the fix should be easy enough, I am just explicitly checking for non-zero integers. I'll handle this edge case as well.
